### PR TITLE
feat: Build Druid from source

### PR DESCRIPTION
**Pull request by labrenbe**
# Description

Successfully ran smoke test with some flakiness that seems to be uncorrelated to this PR.

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
- [ ] Changes are OpenShift compatible
- [ ] All added packages (via microdnf or otherwise) have a comment on why they are added
- [ ] Things not downloaded from Red Hat repositories should be mirrored in the Stackable repository and downloaded from there
- [ ] All packages should have (if available) signatures/hashes verified
- [ ] Add an entry to the CHANGELOG.md file
- [ ] Integration tests ran successfully
```

<details>
<summary>TIP: Running integration tests with a new product image</summary>

The image can be built and uploaded to the kind cluster with the following commands:

```shell
bake --product <product> --image-version <stackable-image-version>
kind load docker-image <image-tagged-with-the-major-version> --name=<name-of-your-test-cluster>
```

See the output of `bake` to retrieve the image tag for `<image-tagged-with-the-major-version>`.
</details>

### CODE REVIEW

**Review by Maleware — CHANGES_REQUESTED** [review]
Please also add a entry to the `Changelog.md`. Other than that, great work :)

**Comment by Maleware** [comment]
Builds succeeded on `arm64`

File: druid/Dockerfile [hunk 1]
```diff
@@ -4,13 +4,76 @@
...
+    curl --fail -L -o /stackable/apache-druid-${PRODUCT}/lib/stax2-api-${STAX2_API}.jar \
+        "https://repo.stackable.tech/repository/packages/stax2-api/stax2-api-${STAX2_API}.jar" && \
+    curl --fail -L -o /stackable/apache-druid-${PRODUCT}/lib/woodstox-core-${WOODSTOX_CORE}.jar \
+        "https://repo.stackable.tech/repository/packages/woodstox-core/woodstox-core-${WOODSTOX_CORE}.jar"
+
+# ===
+# Mitigation for CVE-2021-44228 (Log4Shell)
+# This variable is supported as of Log4j version 2.10 and
+# disables the vulnerable feature
+ENV LOG4J_FORMAT_MSG_NO_LOOKUPS=true
```
Comment (maltesander) [inline druid/Dockerfile:53]: This should go into the main image.

File: conf.py [hunk 2]
```diff
@@ -60,6 +60,7 @@
                 "stax2_api": "4.2.1",
                 "woodstox_core": "6.2.1",
                 "authorizer": "0.5.0",
+                "java-devel": "11",
```
Comment (sbernauer) [inline conf.py:63]: Nit: I think it would be great if we have the attribute `java-devel` right below `java-base`

**Comment by lfrancke** [comment]
Thanks for this! I looked at it and have nothing to add that hasn't already been said.

File: druid/Dockerfile (hunk 1 above)
Comment (Maleware) [reply druid/Dockerfile]: Done https://github.com/stackabletech/docker-images/pull/684/commits/f33bc215feafade57b11ff1727a1c6ae3ef2d1f2

File: druid/Dockerfile [hunk 3]
```diff
@@ -4,15 +4,78 @@
...
+# Further investigation will be needed whether this should also be removed.
+COPY shared/log4shell_1.6.1-log4shell_Linux_x86_64 /bin/log4shell_scanner_x86_64
+COPY shared/log4shell_1.6.1-log4shell_Linux_aarch64 /bin/log4shell_scanner_aarch64
+COPY shared/log4shell_scanner /bin/log4shell_scanner
+RUN /bin/log4shell_scanner s "/stackable/apache-druid-${PRODUCT}"
+# ===
+
+## Final Image
+# hadolint ignore=DL3006
+FROM stackable/image/java-base
```
Comment (sbernauer) [inline druid/Dockerfile:68]: Nit, feel free to ignore
```suggestion
# hadolint ignore=DL3006
FROM stackable/image/java-base as final
```

**Review by sbernauer — DISMISSED** [review]
Thanks!

**Review by maltesander — APPROVED** [review]
LGTM!

**Review by Maleware — APPROVED** [review]

**Pull request merged**